### ISSUE TITLE
In D3, center function on zoom only works with mousewheel, not slider

### ISSUE THREAD

**Issue by kdahlquist** [dondi/GRNsight#468]
This has been noted in a couple of different contexts:

When you zoom in with the range slider, it doesn't zoom to the center. It instead zooms to the top right corner. This is because the center function on zoom only works on mousewheel scrolling, not regular scrolling. @NAnguiano is trying to figure out a way to imitate that behavior for the slider, without needing to do any overly complicated math that may slow it down.

This bug is not in the critical path and can be solved later.

**Comment by eileenchoe** [comment]
Started working on this one too. I got the slider to correctly zoom from the center, but only when the graph itself is centered. There's some more complicated business with d3 group transforms I'm still trying to figure out.

**Comment by eileenchoe** [comment]
http://www.computing.dcu.ie/~dganguly/d3sliderzoom.htm

**Comment by eileenchoe** [comment]
Sorry accidentally closed the issue

**Comment by kdahlquist** [comment]
This is fundamentally about getting the right translate-scale-translate transform for the zoom. The mouse wheel event handling is D3-native and the slider control cannot just "piggyback" on that logic. @eileenchoe can explore the setting of the transform more, plus she can also look at the D3 mouse wheel event handler code itself to see what it's doing to compute the correct transform.

**Comment by eileenchoe** [comment]
https://github.com/d3/d3-zoom/blob/master/src/zoom.js found the code used for D3 zoom for reference. Going to take closer look at it this afternoon

**Comment by eileenchoe** [comment]
I've made pretty good progress on this issue so far. Zoom slider now mimics the behavior of the mousewheel zoom! I have most of the functionality in terms of viewport, centering, d-pad, etc. working with this new change. The last thing I'm working on is to allow the user to move the bounding box when the viewport is fixed. Somehow this was more closely tied with the zoom-wheel.

In development, I had to disable the mouse wheel zooming behavior, because it began to conflict with the zoom slider. Since we are planning to be phasing this out anyways, I've started to rewrite some functions without it. I've been playing around with GRNsight a lot this morning with the mouse wheel disabled, I realize how negatively it had been impacting the usability so I think this will be a great change.
